### PR TITLE
stream: compare modified at

### PR DESCRIPTION
**Pull request by skyzh**

### CODE REVIEW

File: src/stream_pipe.rs [hunk 1]
```diff
@@ -145,20 +145,41 @@ where
...
+
+        let modified_at = modified_at.unwrap();
+
+        if let Some(snapshot_modified_at) = snapshot_modified_at {
+            if let Some(http_modified_at) = http_modified_at {
+                if snapshot_modified_at != http_modified_at {
+                    return Err(Error::PipeError(format!(
+                        "mismatch modified time: http={}, snapshot={}",
+                        http_modified_at, snapshot_modified_at
+                    )));
```
Comment (PhotonQuantum) [inline src/stream_pipe.rs:178]: Why not treat this mismatch as a warning instead of an error?

File: src/stream_pipe.rs [hunk 2]
```diff
@@ -145,20 +145,41 @@ where
...
-            .unwrap();
-            DateTime::parse_from_rfc2822(&header)?.timestamp() as u64
+            http_modified_at
         };
 
+        if modified_at.is_none() {
+            return Err(Error::PipeError("no modified time".to_string()));
+        }
+
+        let modified_at = modified_at.unwrap();
```
Comment (PhotonQuantum) [inline src/stream_pipe.rs:170]:
```rust
let modified_at = modified_at.ok_or_else(||Error::PipeError("no modified time".to_string()))?;
```

File: src/stream_pipe.rs (hunk 1 above)
Comment (skyzh) [reply src/stream_pipe.rs:178]: The file might have been updated after we take a snapshot. We should safely ignore file transfer upon finding mismatch.

**Review by PhotonQuantum — APPROVED** [review]
LGTM

**Comment by skyzh** [comment]
Please release a new version by tagging, and update mirror-docker-unified.

**Pull request merged**